src/storage/src/hummock/store/version.rs

```diff
@@ -82,9 +108,15 @@ impl StagingVersion {
             .iter()
             .filter(move |staging_sst| {
                 *staging_sst.epochs.last().expect("epochs not empty") <= epoch
-                    && filter_single_sst(&staging_sst.sst_info, table_id, key_range)
             })
-            .map(|staging_sst| &staging_sst.sst_info);
+            .flat_map(move |staging_sst| {
```

Comment:
need a related test case for multi state_table_infos for checking `prune_overlap`

Reply:
I have expand our current unit test to cover the newly added code.